### PR TITLE
DCOS CLI Integration

### PR DESCRIPTION
**Pull request by huntc**
This commit provides the changes required in order for the "conduct" CLI to be integrated with the DC/OS CLI such that you can "dcos conduct ..." and the ConductR commands are channelled via DC/OS's admin gateway.

### CODE REVIEW

**Comment by huntc** [comment]
Observational testing:

```
ChristohersMBP2:conductr-cli huntc$ dcos 
Command line utility for the Mesosphere Datacenter Operating
System (DC/OS). The Mesosphere DC/OS is a distributed operating
system built around Apache Mesos. This utility provides tools
for easy management of a DC/OS installation.

Available DC/OS commands:

    auth            Authenticate to DC/OS cluster
    conduct         Lightbend ConductR sub commands. Type 'dcos conduct' to see more.
    config          Manage the DC/OS configuration file
    help            Display help information about DC/OS
    marathon        Deploy and manage applications to DC/OS
    node            Administer and manage DC/OS cluster nodes
    package         Install and manage DC/OS software packages
    service         Manage DC/OS services
    task            Manage DC/OS tasks

Get detailed command description with 'dcos <command> --help'.
ChristohersMBP2:conductr-cli huntc$ 
ChristohersMBP2:conductr-cli huntc$ 
ChristohersMBP2:conductr-cli huntc$ dcos conduct info
ID       NAME        #REP  #STR  #RUN
c14bd62  visualizer     1     0     0
```

**Comment by huntc** [comment]
Failing because of the DC/OS Python library unable to be found.

File: conductr_cli/conduct_main.py
Comment (viktorklang) [inline conductr_cli/conduct_main.py:23]: Shouldn't something as important as the orchestrator API default to https? (perhaps that's for the future?)

Comment (huntc) [reply conductr_cli/conduct_main.py:23]: Agreed. TLS support for the control protocol is planned. Meanwhile our DC/OS integration here will provide TLS support.

File: conductr_cli/conduct_main.py
Comment (markusjura) [inline conductr_cli/conduct_main.py:330]: We are removing here always the first argument. Also in non DCOS mode. Does that makes sense?

File: conductr_cli/conduct_main.py
Comment (markusjura) [inline conductr_cli/conduct_main.py]: For readability reasons I'd would swap the if and else clause:

```
        if vars(args).get('dcos_info'):
            print('Lightbend ConductR sub commands. Type \'dcos conduct\' to see more.')
            exit(0)            
        else:
            parser.print_help()
```

File: conductr_cli/sandbox_run.py
Comment (markusjura) [inline conductr_cli/sandbox_run.py]: Always specify the named argument when passing in a boolean:

```
conduct_request.get(dcos_mode=False, ConductArgs.ip, url, timeout=DEFAULT_HTTP_TIMEOUT)
```

File: conductr_cli/test/conduct_load_test_base.py
Comment (markusjura) [inline conductr_cli/test/conduct_load_test_base.py:74]: Why is this change necessary?

**Review by markusjura — CHANGES_REQUESTED** [review]

**Comment by markusjura** [comment]
> conduct init needs to create a soft link to the `conduct.py` file from `~/.dcos/subcommands/conductr/env/bin/dcos-conduct`

What do you mean by that? We only have a `sandbox init` command but this should not be involved in DCOS mode.

**Comment by huntc** [comment]
> What do you mean by that? We only have a sandbox init command but this should not be involved in DCOS mode.

In order to make the DC/OS CLI aware of the ConductR CLI, you have to tell it. Ordinarily, this is done via `dcos install --cli <package>`, but we don't have a universal package available (yet). So, meanwhile then, `conduct init` will automatically set things up.

I just realised though that we have no `conduct init`, so we'll go with your IM suggestion of a new `conduct setup-dcos`.

File: conductr_cli/conduct_main.py
Comment (huntc) [reply conductr_cli/conduct_main.py:330]: Yes. This is what the previous code was actually doing under the hood.

File: conductr_cli/conduct_main.py
Comment (huntc) [reply conductr_cli/conduct_main.py]: Why would that be more readable?

File: conductr_cli/sandbox_run.py
Comment (huntc) [reply conductr_cli/sandbox_run.py]: ok, but I do then have to specify the other arg names.

File: conductr_cli/test/conduct_load_test_base.py
Comment (huntc) [reply conductr_cli/test/conduct_load_test_base.py:74]: We were previously checking the host header being set up via another mock. The other mock was no longer available due to its function having been replaced. The code should also be much simpler now.

File: conductr_cli/conduct_main.py
Comment (markusjura) [reply conductr_cli/conduct_main.py]: We are saving a "not". In a if / else statement you should never start with the negative, but always with the positive case.

File: conductr_cli/sandbox_run.py
Comment (markusjura) [reply conductr_cli/sandbox_run.py]: Your decision. Not critical.

File: conductr_cli/test/conduct_load_test_base.py
Comment (markusjura) [reply conductr_cli/test/conduct_load_test_base.py:74]: 👍

**Review by markusjura — APPROVED** [review]

File: conductr_cli/conduct_main.py
Comment (markusjura) [reply conductr_cli/conduct_main.py:330]: 👍

File: conductr_cli/conduct_main.py
Comment (huntc) [reply conductr_cli/conduct_main.py]: There's also an argument to show the happy path, but I'll go with it.

**Comment by huntc** [comment]
Waiting on a new DCOS lib release. PR tests will fail here until the following PR is merged and there's a new release of the DCOS lib: https://github.com/dcos/dcos-cli/pull/823

**Comment by markusjura** [comment]
Manual integration test with `http://dcos-master.lightbend.com` was successful.

**Pull request merged**
